core/modules/filters/lookup.js

```diff
@@ -20,10 +22,35 @@ Prepends the prefix to the selected items and returns the specified field value
 Export our filter function
 */
 exports.lookup = function(source,operator,options) {
-	var results = [];
-	source(function(tiddler,title) {
-		results.push(options.wiki.getTiddlerText(operator.operand + title) || operator.suffix || '');
-	});
+	var results = [],
+		suffixes = operator.suffixes || [],
+		defaultSuffix = suffixes[0] ? (suffixes[0][0] || "") : "",
+		indexSuffix = (suffixes[1] && suffixes[1][0] === "index") ? true : false,
+		target;
+	if(operator.operands.length == 2) {
+		target = operator.operands[1]
+	} else {
+		target = indexSuffix ? "0": "text";
+	}
+	if(indexSuffix) {
+		source(function(tiddler,title) {
+			var targetTitle = operator.operands[0] + title;
+			var data = options.wiki.extractTiddlerDataItem(targetTitle,target,defaultSuffix);
+			results.push(data);
+		});
+	} else {
+		source(function(tiddler,title) {
+			var targetTitle = operator.operands[0] + title;
+			var targetTiddler = options.wiki.getTiddler(targetTitle);
+			if(targetTiddler) {
```

Comment:
I believe the issue is here. If the `targetTiddler` does not exist, we need to return the default value if one has been specified. 

I recommend that we add tests though @joshuafontany